aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -437,6 +439,44 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     LOG.info("Successfully renamed table from {} to {}", from, to);
   }
 
+  @Override
+  public org.apache.iceberg.Table registerTable(TableIdentifier identifier, String metadataFileLocation) {
+    Preconditions.checkArgument(isValidIdentifier(identifier),
+        "Table identifier to register is invalid: " + identifier);
+    Preconditions.checkArgument(metadataFileLocation != null && !metadataFileLocation.isEmpty(),
+        "Cannot register an empty metadata file location as a table");
+
+    Map<String, String> tableParameters = ImmutableMap.of(
+        BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+        BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toLowerCase(Locale.ENGLISH),
+        BaseMetastoreTableOperations.METADATA_LOCATION_PROP,
+        metadataFileLocation);
+
+    TableInput tableInput = TableInput.builder()
+        .name(IcebergToGlueConverter.getTableName(identifier, awsProperties.glueCatalogSkipNameValidation()))
+        .tableType(GlueTableOperations.GLUE_EXTERNAL_TABLE_TYPE)
+        .parameters(tableParameters)
+        .build();
+
+    try {
+      glue.createTable(CreateTableRequest.builder()
+          .databaseName(IcebergToGlueConverter.getDatabaseName(identifier,
+              awsProperties.glueCatalogSkipNameValidation()))
+          .tableInput(tableInput)
+          .build());
+    } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
+      glue.updateTable(UpdateTableRequest.builder()
+          .databaseName(IcebergToGlueConverter.getDatabaseName(identifier,
+              awsProperties.glueCatalogSkipNameValidation()))
+          .tableInput(tableInput)
+          .build());
```

Comment:
this should first get the last table version to avoid commit conflict

Reply:
I think glue catalog already handle this internally?

If we call glue.getTable to get the table version first then call glue.updateTable with nextVersionId it will cause concurrentModificationException.

Reply:
Yes this is what I mean. It needs to explicitly pass in the version number of the current version to ensure atomic update.

Reply:
Ack, will update